docs/graphql.md

```diff
@@ -0,0 +1,102 @@
+# GraphQL transactions and spans
+
+**NB:** This document is a work in progress.
+
+Our current approach of splitting and grouping transaction by HTTP method and path fits perfectly for REST-style APIs and the like.
+
+With a GraphQL approach, the client always requests the same endpoint. This means wildly different queries and mutations all will end up in the same transaction group.
+
+To better support GraphQL servers we need to find another way to tell the queries apart.
+
+Example GraphQL query:
+
+```graphql
+{
+  user {
+    id
+    name
+    comments {
+      body
+    }
+  }
+}
+```
+
+**Note:** The Node.js agent already supports GraphQL. This spec is written with that in mind but not necessarily with its implementation as a target result.
+
+## Prefix
+
+To distinguish GraphQL spans from others we prefix them with `GraphQL:`. A span with just the name `User` will be hard to recognise from just that name, whereas `GraphQL:User` is easy to recognise.
+
+## Operation Name
+
+It is common (and [recommended](https://graphql.org/learn/queries/#operation-name)) to provide an _Operation Name_ for queries. Here for example `UserWithName`:
+
+```graphql
+query UserWithComments {
+  user {
+    id
+    name
+    comments {
+      body
+    }
+  }
+}
+```
+
+The point of these are to provide an easy way for the developers, when things go wrong, to pinpoint where exactly they did so.
+
+This name is available on the server too and serves as a great distinguishing key.
+
+Span name examples:
+- `GraphQL:UserWithComments`
+- `GraphQL:UpdateUser`
+
+## Multiple endpoints
+
+An app may serve multiple GraphQL endpoints. To tell them apart we can include the path in the transaction name.
+
+- `GraphQL:UserWithComments (/api/graphql)`
```

Comment:
IMO, graphql is mostly used as a API gateway in most of the companies (Ex: Zalando, Airbnb, GitHub) which means the URL of the endpoint never changes between versions like how REST does. We can safely assume the URL never changes and use the available space in the transaction name for other details.

Reply:
This was a detail that @watson mentioned re: the Node.js implementation.

I don't have much experience but my sense was the same as your's @vigneshshanmugam. Hence the proposed option to enable/disable the path part.